### ISSUE TITLE
Regex string seems to be escaped incorrectly

### ISSUE THREAD

**Issue by andywer** [soldair/node-qrcode#94]
The string used to construct RegExps from seems to contain incorrectly escaped unicode characters / invalid characters, making uglify break the code in our app.

https://github.com/soldair/node-qrcode/blob/df6434054564bbc0dc9877fae59e4dd0ea5d0444/lib/core/regex.js#L3

There should probably be double backslashes to actually pass the escaped unicode codes to the RegExp constructor.
Right now they turn into their actual characters before being consumed by the RegExp constructor:

```
$ node
> '[\u3000-\u303F]'
'[　-〿]'
>
```

So it's kind-of valid for uglify to resolve those escaped unicode chars in the string during build time what unfortunately results in a syntax error:

```
Uncaught SyntaxError: Invalid regular expression: /(?:[ã€€-ã€¿]|[ã€-ã‚Ÿ]|[ã‚ -ãƒ¿]|[ï¼€-ï¿¯]|[ä¸€-é¾¯]|[â˜…-â˜†]|[â†-â†•]|â€»|[â€â€•â€˜â€™â€¥â€¦â€œâ€âˆ¥â‰ ]|[Î‘-Ñ‘]|[Â§Â¨Â±Â´Ã—Ã·])+/: Range out of order in character class
```

Seems more like an escaping issue than an issue with uglify.

**Comment by vigreco** [comment]
Hi @andywer 
this seems a file encoding issue.

Uglifyjs converts unicode chars to UTF-8 but your output file seems to have a different encoding (Central European - Windows 1250 I guess?)
Could you try to convert your minified file to UTF-8 and see if it still raises the error?

I'm not sure that using double backslashes will work here, `\\` will match the character `\` literally. But I'm not a regular expression expert so I could be wrong 🤔 

Perhaps the uglifyjs option `ascii-only=true` could be another solution?

Anyway, we also use Uglifyjs to generate the minified version of `qrcode` and we never encountered that error. Which version of Uglifyjs are you using?

**Comment by andywer** [comment]
Thanks, @vigreco. Good points.

We are using webpack@2.3.3, using uglify-js@2.8.14. The resulting output files are published to AWS S3.

The idea behind the double backslash was to be resolved to a single `\`, resulting in the actual `\uXXXX` unicode escape sequence that it is resolved by the regex engine, not before :)

But yeah, might be a bad charset in the headers. Unfortunately I don't have access to the flawed files anymore, since we already deployed a hotfix, just replacing node-qrcode with another package 🙃

**Comment by vigreco** [comment]
> The idea behind the double backslash was to be resolved to a single \, resulting in the actual \uXXXX unicode escape sequence that it is resolved by the regex engine, not before :)

Ohh I see what you meant now 😄 

Thanks for the info, I'll try to reproduce the issue.

**Comment by vigreco** [comment]
A fix for this is available in `v1.0.0`. Thanks for reporting it!
Feel free to reopen if the issue is still not solved for you.
